test/e2etest/kubetest/setup/setup.go

```diff
@@ -0,0 +1,123 @@
+		return false // path does not exist
+	}
+	ginkgoPath := path.Join(k8sOutputBinDir, "ginkgo")
+	if _, err := os.Stat(ginkgoPath); os.IsNotExist(err) {
+		log.Infof("test utility not ready: %s", ginkgoPath)
+		return false // path does not exist
+	}
+	log.Info("all test utilities were already ready")
+	return true
+}
+
+func downloadKubernetes(k8sVersion string) error {
+	log.Infof("downloading kubernetes v%s", k8sVersion)
+
+	if _, err := os.Stat(config.KubernetesPath); !os.IsNotExist(err) {
+		if err := util.RunCmd("git checkout master", config.KubernetesPath); err != nil {
+			log.Errorf("failed to checkout master branch in %s", config.KubernetesPath, err)
+		}
+	} else {
+		log.Infof("directory %s does not exist", config.KubernetesPath)
+	}
+
+	if err := util.RunCmd("go get -d k8s.io/kubernetes", ""); err != nil {
+		log.Error("failed to go get k8s.io/kubernetes", err)
+	}
+
+	if err := util.RunCmd(fmt.Sprintf("git checkout v%s", k8sVersion), config.KubernetesPath); err != nil {
+		log.Error(err)
+	}
+	return nil
+}
+
+func compileOrGetTestUtilities(k8sVersion string) error {
+	k8sTestBinariesVersionURL := fmt.Sprintf("https://storage.googleapis.com/kubernetes-release/release/v%s/kubernetes-test-%s-amd64.tar.gz", k8sVersion, runtime.GOOS)
+	resp, err := http.Get(k8sTestBinariesVersionURL)
+
+	if err != nil || resp.StatusCode != http.StatusOK || (runtime.GOOS != "linux" && runtime.GOOS != "darwin") {
+		log.Info("no precompiled kubernetes test binaries available, or operating system is not linux/darwin, build e2e.test and ginko")
+		_ = util.RunCmd("make WHAT=test/e2e/e2e.test", config.KubernetesPath)
+		_ = util.RunCmd("make WHAT=vendor/github.com/onsi/ginkgo/ginkgo", config.KubernetesPath)
+	} else if resp.StatusCode == http.StatusOK {
+		log.Infof("precompiled kubernetes test binaries available, download kubernetes-test-linux-amd64 for v%s", k8sVersion)
+		k8sTestBinariesTarPath, err := util.DownloadFile(k8sTestBinariesVersionURL, config.TmpDir)
+		if err != nil {
+			return err
+		}
+		if err := archiver.Unarchive(k8sTestBinariesTarPath, config.TmpDir); err != nil {
+			return err
+		}
+		var extractedDirPath string = filepath.Join(config.TmpDir, "kubernetes/test/bin")
+		_ = os.MkdirAll(filepath.Dir(k8sOutputBinDir), os.FileMode(0777))
+		if err := os.Rename(extractedDirPath, k8sOutputBinDir); err != nil {
+			return err
+		}
+
+		if runtime.GOOS == "linux" {
+			log.Info("Install glibc to run precompiled ginkgo and e2e.test binaries")
+			_ = util.RunCmd("apk --no-cache add ca-certificates wget", "")
+			_ = util.RunCmd("wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub", "")
+			_ = util.RunCmd("wget --quiet https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.29-r0/glibc-2.29-r0.apk", "")
+			_ = util.RunCmd("apk add glibc-2.29-r0.apk", "")
+		}
+	}
+
+	log.Info("get k8s examples")
+	_ = util.RunCmd("go get -u k8s.io/examples > /dev/null", "")
```

Comment:
why are you running `go get ..` into `dev/null`?

Reply:
this command always returns a false positive error

Reply:
ok but you also ignore the error on golang level, so `/dev/null` has the same functionality  as `_=`?

Reply:
now stderr is returned as separate return value and is checked whether it contains text like "no Go files in", which comes along with exit code 1 as false positive